unison-cli/src/Unison/CommandLine/InputPatterns.hs

```diff
@@ -207,14 +217,82 @@ import Unison.Project
     branchWithOptionalProjectParser,
   )
 import Unison.Project.Util (ProjectContext (..), projectContextFromPath)
-import Unison.Syntax.HashQualified qualified as HQ (parseText)
-import Unison.Syntax.Name qualified as Name (parseText, unsafeParseText)
-import Unison.Syntax.NameSegment qualified as NameSegment (renderParseErr, segmentP)
+import Unison.Reference qualified as Reference
+import Unison.Referent qualified as Referent
+import Unison.Server.Backend (ShallowListEntry (..))
+import Unison.Server.Backend qualified as Backend
+import Unison.Server.SearchResult (SearchResult)
+import Unison.Server.SearchResult qualified as SR
+import Unison.ShortHash (ShortHash)
+import Unison.Syntax.HashQualified qualified as HQ (parseText, toText)
+import Unison.Syntax.Name qualified as Name (parseTextEither, toText)
+import Unison.Syntax.NameSegment qualified as NameSegment
 import Unison.Util.ColorText qualified as CT
 import Unison.Util.Monoid (intercalateMap)
 import Unison.Util.Pretty qualified as P
 import Unison.Util.Pretty.MegaParsec (prettyPrintParseError)
 
+-- |
+--
+--  __FIXME__: Don’t hardcode this
+schLength :: Int
+schLength = 10
+
+formatStructuredArgument :: StructuredArgument -> Text
+formatStructuredArgument = \case
```

Comment:
Not quite. This is used for printing the list of numbered args in the first place, before there is even a consumer, so that will stick around.

Currently, it is also used for history, and that will still be true when this PR lands. The handling of number expansion and history is currently in one place. It would be a separate change to allows the `InputPattern`s to make decisions about those things.

Reply:
> This is used for printing the list of numbered args in the first place, before there is even a consumer, so that will stick around.

You mean in `debug.numberedArgs`? 👍 

> Currently, it is also used for history, and that will still be true when this PR lands. The handling of number expansion and history is currently in one place. It would be a separate change to allows the `InputPattern`s to make decisions about those things.

Ah right, ok.